### PR TITLE
chore: add CSI tests for e2e-qemu

### PR DESCRIPTION
**Pull request by frezbo**
Add tests for using rook as CSI for e2e-qemu

Signed-off-by: Noel Georgi <git@frezbo.dev>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4841)

### CODE REVIEW

File: hack/test/e2e.sh [hunk 1]
```diff
@@ -211,3 +211,13 @@ function build_registry_mirrors {
...
 }
+
+function run_csi_tests {
+  git clone --depth=1 --single-branch --branch v1.8.2 https://github.com/rook/rook.git "${TMP}/rook"
+  pushd "${TMP}/rook/deploy/examples"
+  ${KUBECTL} create -f crds.yaml -f common.yaml -f operator.yaml
+  ${KUBECTL} create -f cluster.yaml
+  ${KUBECTL} --namespace rook-ceph wait --timeout=600s --for=jsonpath='{.status.ceph.health}=HEALTH_OK' cephclusters.ceph.rook.io/rook-ceph
+  ${KUBECTL} create -f csi/rbd/storageclass.yaml
+  ${KUBESTR} fio --storageclass rook-ceph-block --size 10G
```
Comment (smira) [inline hack/test/e2e.sh:222]: :heart:

File: hack/test/e2e-qemu.sh [hunk 2]
```diff
@@ -125,6 +139,7 @@ case "${TEST_MODE:-default}" in
     get_kubeconfig
     run_talos_integration_test
     run_kubernetes_integration_test
+    ${CSI_STORAGE_CLASS_FLAG}
```
Comment (smira) [inline hack/test/e2e-qemu.sh:142]: this also could have used `WITH_TEST` variable I introduced in #4884

**Review by smira — APPROVED** [review]

File: .drone.jsonnet [hunk 3]
```diff
@@ -403,6 +403,16 @@ local integration_qemu_encrypted_vip = Step("e2e-encrypted-vip", target="e2e-qem
...
 });
 
+local integration_qemu_csi = Step("e2e-csi", target="e2e-qemu", privileged=true, depends_on=[load_artifacts], environment={
+        "IMAGE_REGISTRY": local_registry,
+        "QEMU_WORKERS": "3",
+        "QEMU_CPUS_WORKERS": "4",
+        "QEMU_MEMORY_WORKERS": "5120",
+        "QEMU_EXTRA_DISKS": "1",
+        "QEMU_EXTRA_DISKS_SIZE": "12288",
+        "WITH_TEST": "run_csi_tests",
```
Comment (smira) [inline .drone.jsonnet:413]: do you want to run short integration test here? as it runs anyways before the Ceph is deployed, it doesn't give much new information?

Comment (frezbo) [reply .drone.jsonnet:414]: fixed

**Pull request merged**